src/focus.service.ts

```diff
@@ -362,46 +332,35 @@ export class FocusService {
       return;
     }
 
-    this.rescroll(next, scrollSpeed, this.root);
-
     const attached = selected && isNodeAttached(selected, this.root);
 
-    this.parents = [];
-
     if (attached) {
       // Find the common ancestor of the next and currently selected element.
       // Remove selected classes in the current subtree, and add selected
       // classes in the other subtree. Trigger focus changes on every
       // element that we touch.
-      selected.blur();
-
       const common = getCommonAncestor(next, selected);
-      selected.classList.remove(cssClass.direct);
```

Comment:
That seems like an extremely difficult goal to pursue, I would very much encourage rethinking how focus states are communicated over trying to distinguish between causes of focus states. Browsers don't 'know' about focus sources, this seems like a rabbit hole of edge cases. What happens if I click then navigate with a keyboard? What is autofocused considered? If the focus is transferred programmatically, what happens? How does it interact with screen readers and other accessibility software?